core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -566,7 +566,7 @@ private[spark] class Executor(
           val (accums, accUpdates) = collectAccumulatorsAndResetStatusOnFailure(taskStartTimeNs)
           // Here and below, put task metric peaks in a WrappedArray to expose them as a Seq
           // without requiring a copy.
-          val metricPeaks = WrappedArray.make(metricsPoller.getTaskMetricPeaks(taskId))
+          val metricPeaks = metricsPoller.getTaskMetricPeaks(taskId)
```

Comment:
Sean, I think we should fix RowEncoder too if we shouldn't use WrappedArary:

https://github.com/apache/spark/blob/b917a6593dc969b9b766259eb8cbbd6e90e0dc53/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala#L294-L299

Out of curiosity, what will we use as the replacement for WrappedArray? I'm asking this as it might affect users directly:

```scala

scala> spark.range(1).selectExpr("array(1)").collect()
res0: Array[org.apache.spark.sql.Row] = Array([WrappedArray(1)])
```